<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>DBppGuiAppl15</class>
 <widget class="QMainWindow" name="DBppGuiAppl15">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1382</width>
    <height>755</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>DAMBREAK++ Simulator</string>
  </property>
  <widget class="QWidget" name="centralWidget">
   <widget class="QGroupBox" name="AlgoSelect">
    <property name="geometry">
     <rect>
      <x>30</x>
      <y>50</y>
      <width>211</width>
      <height>251</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <pointsize>8</pointsize>
     </font>
    </property>
    <property name="title">
     <string>Algporithm Selection</string>
    </property>
    <widget class="QWidget" name="verticalLayoutWidget">
     <property name="geometry">
      <rect>
       <x>20</x>
       <y>30</y>
       <width>191</width>
       <height>48</height>
      </rect>
     </property>
     <layout class="QVBoxLayout" name="verticalLayout">
      <item>
       <widget class="QLabel" name="reconstr_label">
        <property name="text">
         <string>Reconstruction Algorithm</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QComboBox" name="reconstr_comboBox">
        <property name="editable">
         <bool>true</bool>
        </property>
        <property name="currentText">
         <string>MUSCL Slope Limiter</string>
        </property>
       </widget>
      </item>
     </layout>
    </widget>
    <widget class="QWidget" name="verticalLayoutWidget_2">
     <property name="geometry">
      <rect>
       <x>20</x>
       <y>100</y>
       <width>191</width>
       <height>48</height>
      </rect>
     </property>
     <layout class="QVBoxLayout" name="verticalLayout_2">
      <item>
       <widget class="QLabel" name="numalgo_label">
        <property name="text">
         <string>Numerical Method</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QComboBox" name="numalgo_combo">
        <property name="editable">
         <bool>true</bool>
        </property>
        <property name="currentText">
         <string/>
        </property>
        <property name="currentIndex">
         <number>-1</number>
        </property>
       </widget>
      </item>
     </layout>
    </widget>
    <widget class="QWidget" name="verticalLayoutWidget_9">
     <property name="geometry">
      <rect>
       <x>20</x>
       <y>170</y>
       <width>191</width>
       <height>48</height>
      </rect>
     </property>
     <layout class="QVBoxLayout" name="verticalLayout_9">
      <item>
       <widget class="QLabel" name="rhsalgo_label">
        <property name="text">
         <string>Physical Algorithm</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QComboBox" name="rhsalgo_combo">
        <property name="currentText">
         <string/>
        </property>
       </widget>
      </item>
     </layout>
    </widget>
   </widget>
   <widget class="QGroupBox" name="Simulation">
    <property name="geometry">
     <rect>
      <x>270</x>
      <y>50</y>
      <width>291</width>
      <height>251</height>
     </rect>
    </property>
    <property name="title">
     <string>Parameters of run</string>
    </property>
    <widget class="QWidget" name="verticalLayoutWidget_4">
     <property name="geometry">
      <rect>
       <x>20</x>
       <y>100</y>
       <width>121</width>
       <height>45</height>
      </rect>
     </property>
     <layout class="QVBoxLayout" name="verticalLayout_4">
      <item>
       <widget class="QLabel" name="nbIter">
        <property name="text">
         <string>Nb Iterations</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QSpinBox" name="iterations">
        <property name="minimum">
         <number>1</number>
        </property>
       </widget>
      </item>
     </layout>
    </widget>
    <widget class="QWidget" name="verticalLayoutWidget_5">
     <property name="geometry">
      <rect>
       <x>170</x>
       <y>30</y>
       <width>111</width>
       <height>48</height>
      </rect>
     </property>
     <layout class="QVBoxLayout" name="verticalLayout_5">
      <item>
       <widget class="QLabel" name="phi1_label">
        <property name="text">
         <string>Phi1 Value</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QDoubleSpinBox" name="_phi1">
        <property name="minimum">
         <double>1.000000000000000</double>
        </property>
        <property name="maximum">
         <double>100.000000000000000</double>
        </property>
        <property name="singleStep">
         <double>10.000000000000000</double>
        </property>
        <property name="value">
         <double>10.000000000000000</double>
        </property>
       </widget>
      </item>
     </layout>
    </widget>
    <widget class="QWidget" name="verticalLayoutWidget_6">
     <property name="geometry">
      <rect>
       <x>170</x>
       <y>100</y>
       <width>111</width>
       <height>48</height>
      </rect>
     </property>
     <layout class="QVBoxLayout" name="verticalLayout_6">
      <item>
       <widget class="QLabel" name="phi0_label">
        <property name="cursor">
         <cursorShape>SizeHorCursor</cursorShape>
        </property>
        <property name="text">
         <string>Phi0 Value</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QDoubleSpinBox" name="_phi0">
        <property name="minimum">
         <double>0.500000000000000</double>
        </property>
        <property name="maximum">
         <double>100.000000000000000</double>
        </property>
       </widget>
      </item>
     </layout>
    </widget>
    <widget class="QWidget" name="verticalLayoutWidget_7">
     <property name="geometry">
      <rect>
       <x>170</x>
       <y>170</y>
       <width>111</width>
       <height>51</height>
      </rect>
     </property>
     <layout class="QVBoxLayout" name="verticalLayout_7">
      <item>
       <widget class="QLabel" name="shocloc_label">
        <property name="text">
         <string>Shock Location</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QDoubleSpinBox" name="_shockloc">
        <property name="minimum">
         <double>0.000000000000000</double>
        </property>
        <property name="maximum">
         <double>500.000000000000000</double>
        </property>
        <property name="value">
         <double>0.000000000000000</double>
        </property>
       </widget>
      </item>
     </layout>
    </widget>
    <widget class="QWidget" name="verticalLayoutWidget_8">
     <property name="geometry">
      <rect>
       <x>20</x>
       <y>170</y>
       <width>121</width>
       <height>45</height>
      </rect>
     </property>
     <layout class="QVBoxLayout" name="verticalLayout_8">
      <item>
       <widget class="QLabel" name="nbgridPts_label">
        <property name="text">
         <string>Nb Grid Points</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QSpinBox" name="_nbgridpts">
        <property name="minimum">
         <number>1</number>
        </property>
        <property name="maximum">
         <number>1000</number>
        </property>
        <property name="value">
         <number>100</number>
        </property>
       </widget>
      </item>
     </layout>
    </widget>
    <widget class="QWidget" name="verticalLayoutWidget_3">
     <property name="geometry">
      <rect>
       <x>20</x>
       <y>30</y>
       <width>121</width>
       <height>51</height>
      </rect>
     </property>
     <layout class="QVBoxLayout" name="verticalLayout_3">
      <item>
       <widget class="QLabel" name="cflNumlabel">
        <property name="text">
         <string>CFL Number</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QDoubleSpinBox" name="cflNumSpinBox">
        <property name="maximum">
         <double>1.000000000000000</double>
        </property>
        <property name="singleStep">
         <double>0.100000000000000</double>
        </property>
        <property name="value">
         <double>0.600000000000000</double>
        </property>
       </widget>
      </item>
     </layout>
    </widget>
   </widget>
   <widget class="QTextEdit" name="_msgText">
    <property name="enabled">
     <bool>true</bool>
    </property>
    <property name="geometry">
     <rect>
      <x>30</x>
      <y>310</y>
      <width>831</width>
      <height>191</height>
     </rect>
    </property>
   </widget>
   <widget class="QGroupBox" name="data_groupBox">
    <property name="geometry">
     <rect>
      <x>590</x>
      <y>50</y>
      <width>271</width>
      <height>251</height>
     </rect>
    </property>
    <property name="title">
     <string>Simulation Data</string>
    </property>
    <widget class="QWidget" name="verticalLayoutWidget_11">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>30</y>
       <width>191</width>
       <height>48</height>
      </rect>
     </property>
     <layout class="QVBoxLayout" name="verticalLayout_11">
      <item>
       <widget class="QLabel" name="label">
        <property name="text">
         <string>DamBreakData</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QComboBox" name="discrdata_combo"/>
      </item>
     </layout>
    </widget>
    <widget class="QWidget" name="verticalLayoutWidget_10">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>100</y>
       <width>160</width>
       <height>51</height>
      </rect>
     </property>
     <layout class="QVBoxLayout" name="verticalLayout_10">
      <item>
       <widget class="QLabel" name="label_2">
        <property name="text">
         <string>Manning Coeff</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QDoubleSpinBox" name="doubleSpinBox_2"/>
      </item>
     </layout>
    </widget>
   </widget>
  </widget>
  <widget class="QToolBar" name="mainToolBar">
   <attribute name="toolBarArea">
    <enum>TopToolBarArea</enum>
   </attribute>
   <attribute name="toolBarBreak">
    <bool>false</bool>
   </attribute>
  </widget>
  <widget class="QStatusBar" name="statusBar"/>
  <widget class="QMenuBar" name="menuBar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>1382</width>
     <height>21</height>
    </rect>
   </property>
   <widget class="QMenu" name="menuFile">
    <property name="title">
     <string>File</string>
    </property>
    <addaction name="action_Exit"/>
   </widget>
   <widget class="QMenu" name="menuView">
    <property name="title">
     <string>View</string>
    </property>
   </widget>
   <widget class="QMenu" name="menu_Simulation">
    <property name="title">
     <string>&amp;Simulation</string>
    </property>
    <addaction name="actioninitialize"/>
    <addaction name="action_run"/>
    <addaction name="action_doOneStep"/>
   </widget>
   <addaction name="menuFile"/>
   <addaction name="menuView"/>
   <addaction name="menu_Simulation"/>
  </widget>
  <action name="action_Exit">
   <property name="text">
    <string>&amp;Exit</string>
   </property>
  </action>
  <action name="action_run">
   <property name="text">
    <string>&amp;run</string>
   </property>
  </action>
  <action name="action_doOneStep">
   <property name="text">
    <string>&amp;doOneStep</string>
   </property>
   <property name="shortcut">
    <string>Shift+S, 1</string>
   </property>
  </action>
  <action name="actioninitialize">
   <property name="text">
    <string>&amp;initialize</string>
   </property>
  </action>
 </widget>
 <layoutdefault spacing="6" margin="11"/>
 <resources/>
 <connections>
  <connection>
   <sender>cflNumSpinBox</sender>
   <signal>valueChanged(double)</signal>
   <receiver>DBppGuiAppl15</receiver>
   <slot>setCFLValue(double)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>331</x>
     <y>153</y>
    </hint>
    <hint type="destinationlabel">
     <x>405</x>
     <y>71</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>_shockloc</sender>
   <signal>valueChanged(double)</signal>
   <receiver>DBppGuiAppl15</receiver>
   <slot>setShockLoc(double)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>471</x>
     <y>291</y>
    </hint>
    <hint type="destinationlabel">
     <x>519</x>
     <y>338</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>_phi1</sender>
   <signal>valueChanged(double)</signal>
   <receiver>DBppGuiAppl15</receiver>
   <slot>setUpstream(double)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>496</x>
     <y>149</y>
    </hint>
    <hint type="destinationlabel">
     <x>695</x>
     <y>76</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>_phi0</sender>
   <signal>valueChanged(double)</signal>
   <receiver>DBppGuiAppl15</receiver>
   <slot>setDownstream(double)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>518</x>
     <y>222</y>
    </hint>
    <hint type="destinationlabel">
     <x>694</x>
     <y>49</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>iterations</sender>
   <signal>valueChanged(int)</signal>
   <receiver>DBppGuiAppl15</receiver>
   <slot>setNbIterationsMax(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>307</x>
     <y>216</y>
    </hint>
    <hint type="destinationlabel">
     <x>230</x>
     <y>55</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>numalgo_combo</sender>
   <signal>currentTextChanged(QString)</signal>
   <receiver>DBppGuiAppl15</receiver>
   <slot>setActiveNumericalMethod(QString)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>55</x>
     <y>219</y>
    </hint>
    <hint type="destinationlabel">
     <x>56</x>
     <y>575</y>
    </hint>
   </hints>
  </connection>
 </connections>
 <slots>
  <slot>runSimulation()</slot>
  <slot>setDamBreakData(QString)</slot>
  <slot>Msg2Write(Qstring)</slot>
  <slot>textMsg2Write(Qstring)</slot>
  <slot>setCFLValue(double)</slot>
  <slot>setShockLoc(double)</slot>
  <slot>setUpstream(double)</slot>
  <slot>setDownstream(double)</slot>
  <slot>setNbIterationsMax(int)</slot>
  <slot>setActiveNumericalMethod(QString)</slot>
 </slots>
</ui>
